src/NBomber/Domain/Scenario.fs

```diff
@@ -215,7 +220,8 @@ let applySettings (settings: ScenarioSetting list) (scenarios: Scenario list) =
         { scenario with LoadTimeLine = timeLine.LoadTimeLine
                         WarmUpDuration = getWarmUpDuration(settings)
                         PlanedDuration = timeLine.ScenarioDuration
-                        CustomSettings = settings.CustomSettings |> Option.defaultValue "" }
+                        CustomSettings = settings.CustomSettings |> Option.defaultValue ""
+                        GetStepsOrder = fun () -> getStepsOrder(scenario, settings) }
```

Comment:
I think it could be written like
settings.StepOrder |> Option.defaultValue(scenario.GetStepsOrder())